simulation/traffic_simulator/include/traffic_simulator/helper/helper.hpp

```diff
@@ -89,6 +92,26 @@ geometry_msgs::msg::Vector3 constructRPYfromQuaternion(geometry_msgs::msg::Quate
 geometry_msgs::msg::Pose constructPose(
   double x, double y, double z, double roll, double pitch, double yaw);
 
+/**
+ * @brief helper function for creating vector without duplicates, with preserved order
+ *
+ * @param vector input std::vector
+ * @return new std::vector without duplicates and with relative order preserved
+ */
+template <typename T>
+std::vector<T> getUniqueValues(std::vector<T> input_vector)
```

Comment:
How about using const reference.

```suggestion
std::vector<T> getUniqueValues(const std::vector<T> & input_vector)
```